docusaurus/docs/React/custom-code-examples/channel-search.mdx

```diff
@@ -0,0 +1,313 @@
+
+Let's allow the query to search for `channels` in addition to the default query of just users by setting the `searchForChannels` prop. You'll notice that all props for `ChannelSearch`
+are drilled to the component via the `additionalChannelSearchProps` prop on `ChannelList`. With just this one additional prop, the search functionality is fully functional and expanded from just the default search of users.
+
+```jsx
+<ChannelList
+    showChannelSearch
+    additionalChannelSearchProps={{searchForChannels: true }}
+/>
+```
+
+### Step 2 - Customizing the DropdownContainer
+
+Since we've enabled the search to also search for channels, let's customize the dropdown results container to fit a design that separates the result by type.
+
+```jsx
+  const CustomDropdown = (props: DropdownContainerProps) => {
+    const { results, focusedUser, selectResult, SearchResultItem } = props;
+
+    let items: ChannelOrUserResponse[] = results.filter(x => x.cid);
+    let users: ChannelOrUserResponse[] = results.filter(x => !x.cid);
+
+    return (
+      <div>
+        <p>Channels</p>
+        {!items.length && <p>No Channels...</p>}
+        {items.map((result: ChannelOrUserResponse, index: number) => (
+          <SearchResultItem
+            focusedUser={focusedUser}
+            index={index}
+            key={index}
+            result={result}
+            selectResult={selectResult}
+          />
+        ))}
+        <p>Users</p>
+        {!users.length && <p>No Users...</p>}
+        {users.map((result: ChannelOrUserResponse, index: number) => (
+          <SearchResultItem
+            focusedUser={focusedUser}
+            index={index}
+            key={index}
+            result={result}
+            selectResult={selectResult}
+          />
+        ))}
+      </div>
+    );
+  };
+```
+
+```jsx
+const DropDown = (props: DropdownContainerProps) => <CustomDropdown {...props} />;
+
+<ChannelList
+    showChannelSearch
+    additionalChannelSearchProps={
+        {
+          searchForChannels: true,
+          DropdownContainer: DropDown
+        }
+    }
+/>
+```
+
+### Step 3 - Customizing the Result Component
+
+The separation by the type of result, is a huge UI improvement for users, but the above code is still using the default component for each result. Let's create our own component with more information. We've also added a Type Guard so we know what type we're displaying.
+
+```jsx
+  const isChannelOrUserResponse = (
+    channel: Channel | UserResponse,
+  ): channel is Channel =>
+    (channel as Channel).cid != null;
+
+  const CustomResultItem = (props: SearchResultItemProps) => {
+    const { focusedUser, index, result, selectResult } = props;
+
+    const focused = focusedUser === index;
+
+    if (isChannelOrUserResponse(result)) {
+      const channel = result;
+      const members = channel?.data?.member_count;
```

Comment:
let's destructure member_count off this instead of creating a new members reference